novawallet/Modules/TransactionHistory/Model/TransactionHistoryViewModelFactory.swift

```diff
@@ -72,22 +77,67 @@ final class TransactionHistoryViewModelFactory {
 
         let icon = txType == .incoming ? R.image.iconIncomingTransfer() : R.image.iconOutgoingTransfer()
         let imageViewModel = icon.map { StaticImageViewModel(image: $0) }
-        let subtitle = R.string.localizable.transferTitle(preferredLanguages: locale.rLanguages)
-        let peerAddress = (data.sender == address ? data.receiver : data.sender) ?? data.sender
+        let amountDetails = amountDetails(price: balance.price, time: time, locale: locale)
+        let itemTitleWithSubtitle = createTransactionItemTitleWithSubtitle(
+            data: data,
+            address: address,
+            txType: txType,
+            locale: locale
+        )
 
         return TransactionItemViewModel(
             identifier: data.identifier,
             timestamp: data.timestamp,
-            title: peerAddress,
-            subtitle: subtitle,
-            time: balance.price ?? "",
+            title: itemTitleWithSubtitle.title,
+            subtitle: itemTitleWithSubtitle.subtitle,
             amount: balance.amount,
+            amountDetails: amountDetails,
             type: txType,
             status: data.status.walletValue,
             imageViewModel: imageViewModel
         )
     }
 
+    private func createTransactionItemTitleWithSubtitle(
+        data: TransactionHistoryItem,
+        address: AccountAddress,
+        txType: TransactionType,
+        locale: Locale
+    ) -> TitleWithSubtitleViewModel {
+        let peerAddress = (data.sender == address ? data.receiver : data.sender) ?? data.sender
+        let title = R.string.localizable.transferTitle(preferredLanguages: locale.rLanguages)
+
+        let subtitle = txType == .incoming ?
+            R.string.localizable.walletHistoryTransferIncomingDetails(
+                peerAddress,
+                preferredLanguages: locale.rLanguages
+            ) :
+            R.string.localizable.walletHistoryTransferOutgoingDetails(
+                peerAddress,
+                preferredLanguages: locale.rLanguages
+            )
+
+        return .init(title: title, subtitle: subtitle)
+    }
+
+    private func createEvmContractCallTitleWithSubtitle(
+        data: TransactionHistoryItem,
+        locale: Locale
+    ) -> TitleWithSubtitleViewModel {
+        let title = R.string.localizable.evmContractCall(preferredLanguages: locale.rLanguages)
+
+        if let evmContractFunctionName = data.evmContractFunctionName,
+           !evmContractFunctionName.lowercased().contains("transfer") {
```

Comment:
Let's create an extension for this check because are will reuse it on details screen. For example, ```hasAmbigiousFunctionName```